webpack/assets/javascripts/react_app/components/notifications/drawer/notificationGroup.js

```diff
@@ -39,9 +40,17 @@ export default (
               onClickedLink={onClickedLink}
               key={notification.id}
               notification={notification}
-              onMarkAsRead={onMarkAsRead.bind(this, group)}
+              onMarkAsRead={onMarkAsRead.bind(this, group, [notification.id])}
```

Comment:
consider the situation where you have a group with 200 notifications, that you want to mark as seen.
if you have pagination, you dont have the ids of all of them, and even if you did making seperate requests would be in-efficient (one request would also make a very nasty sql query).

it would be better to handle things simply and directly - passing the group name to the server and let it do the heavy lifting. for optimistic concurrency, the reducer can handle the notifications that are loaded in the ui state.

if we want to support pagination - there is no other way.